### DISCUSSION TITLE
an exe to get file path

### DISCUSSION THREAD

**Discussion by snow-bean** [pyinstaller/pyinstaller#7281] [Help] [answered]
when I try to copy my exe to another path  to get all file name ,  after finished running the exe I get an output , but  the output is not the file name in the path ,where did I wrong ?
my code as below:

import glob
path1=glob.glob('./*')
print(path1)
file1=open('./test.txt','w')
file1.writelines(x+'\n' for x in path1)
file1.close()

 I use PyInstaller to make an exe , and I want use this exe to get filename at a random path.
where did I wrong ?

**Comment by rokm** [comment, marked as answer]
You're using relative paths, which means you are writing the file in your current working directory, which, depending on how you run the executable, may or may not be the same directory that the exe is in. Use `os.path.dirname(__file__)` or `os.path(sys.executable)` (if this is onefile executable) to anchor your relative paths to the executable's location.

**Reply by snow-bean** [reply]
thanks for your reply , I could not have solved this problem so quickly without your help .thanks again!
